### PR TITLE
Bump JSON-RPC

### LINKED ISSUE #10072: JSONRPC charset attribute of Content-type header is case sensitive
- **Parity Ethereum version**: v2.3.0-unstable-665e765cd-20181210/x86_64-linux-gnu/rustc1.31.0
- **Operating system**: Ubuntu 18.04
- **Installation**: built from source
- **Fully synchronized**: yes
- **Network**: mtihani (ellaism progpow test net)
- **Restarted**: yes

**Actual**

I setup a Parity node and tried to connect with a local copy of MyEtherWallet.  The request to JSONRPC endpoint always fails with HTTP error 415 (Unsupported Media Type).  Response body says `Supplied content type is not allowed. Content-Type: application/json is required`

This is the request exported from chrome dev tools in curl format, note the uppercase charset attribute:

> curl 'http://192.168.0.93:8545/' -H 'Accept: application/json, text/plain, */*' -H 'Origin: null' -H 'User-Agent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_14_1) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/71.0.3578.98 Safari/537.36' -H 'Content-Type: application/json; charset=UTF-8' --data-binary '{"method":"eth_blockNumber","id":"1577adb21150879672050dc036f81551","jsonrpc":"2.0"}' --compressed

**Expected**

The same request with the charset attribute in lowercase works fine.  I believe the expected behaviour is that the request should work even with uppercase charset attribute.

**Work around**

I had to setup an nginx proxy to rewrite the _Content-Type_ header to make MyEtherWallet work.

```
upstream jsonrpc {
    server 127.0.0.1:8545;
}

server {
  ...
  location / {
     proxy_pass http://jsonrpc;
     proxy_http_version 1.1;
     proxy_set_header content-type "application/json";
  }
}
```

### PR DESCRIPTION
**Pull request by tomusdrw**
Fixes #10072, kudos @seunlanlege

### CODE REVIEW

File: Cargo.lock [hunk 1]
```diff
@@ -1748,7 +1748,7 @@ dependencies = [
 [[package]]
 name = "jsonrpc-core"
 version = "9.0.0"
-source = "git+https://github.com/paritytech/jsonrpc.git?branch=parity-2.2#f8a54f46f7f1d68b4e7899ca1e929803bf966a5b"
+source = "git+https://github.com/paritytech/jsonrpc.git?branch=parity-2.2#84360bf1d03821b8fd669a6a95fcdf847aaf6cca"
```
Comment (niklasad1) [inline Cargo.lock]: why not a new branch called `parity-2.3` in the `jsonrpc` crate?

**Comment by 5chdn** [comment]
Restarted CI

**Comment by 5chdn** [comment]
lol it fails because of HTTP status casing:

```
left: `"HTTP/1.1 200 OK"`, right: `"HTTP/1.1 200 Ok"
```

File: Cargo.lock (hunk 1 above)
Comment (tomusdrw) [reply Cargo.lock]: There are no significant changes, it's pretty much just a bug fix. We usually create a new branch whenever there are some breaking changes.

Comment (niklasad1) [reply Cargo.lock]: Ok, got it :+1:

**Pull request merged**